core/src/library.rs

```diff
@@ -90,6 +90,21 @@ impl<'gc> Avm2ClassRegistry<'gc> {
         movie: Arc<SwfMovie>,
         symbol: CharacterId,
     ) {
+        if let Some(old) = self.class_map.get(&class_object) {
+            // We should never have multiple symbols in the same movie pointing the same class
+            if Arc::ptr_eq(&movie, &old.0) {
+                panic!(
```

Comment:
If this is theoretically possible through a malicious SWF, wouldn't a warning or error be better suited here?